packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -884,6 +885,72 @@ export const toCustomProperties = ( tree, blockSelectors ) => {
 	return ruleset;
 };
 
+const getFontFaceDeclarations = ( tree ) => {
+	const fonts = tree?.settings?.typography?.fontFamilies;
+	let ruleset = '';
+
+	if ( ! fonts ) {
+		return ruleset;
+	}
+
+	const themeFileURIs = tree?._links?.[ 'wp:theme-file' ] ?? [];
+
+	// Iterate over main origins (theme, custom, default)
```

Comment:
Nit: It could be helpful to include a comment here explaining what this block of code does.

Presumably it's generating @font-face declarations so an example of the generated output would help folks to conceptualise this.

Reply:
:+1: Added here: https://github.com/WordPress/gutenberg/pull/65019/commits/b69ae26276c3db6519fc790d1415500373363bfc